tests/test_tm_data_types.py

```diff
@@ -91,50 +83,47 @@ def test_parallel():
             y_offset=0.0,
             y_position=0.0,
         )
-        waveform_info[waveform_path] = waveform
+        waveform_info[waveform_path.as_posix()] = waveform
 
     write_files_in_parallel(list(waveform_info.keys()), list(waveform_info.values()))
 
     if read_info := read_files_in_parallel(list(waveform_info.keys())):
         for file_path, waveform in read_info:
             assert np.array_equal(waveform.y_axis_values, waveform_info[file_path].y_axis_values)
-            os.remove(file_path)
     else:
         raise IOError("No Files written/read.")
 
 
-def test_wfm():
+@pytest.mark.parametrize(
+    ("waveform_type", "waveform_meta_info"),
+    [(AnalogWaveform, AnalogWaveformMetaInfo)],
+)
+def test_wfm(
+    waveform_type: Type[Waveform], waveform_meta_info: Type[WaveformMetaInfo], tmp_path: Path
+) -> None:
     """Check that a saved waveform is in a format that is readable by the oscilloscope."""
-    waveform_type_meta_info = [AnalogWaveformMetaInfo]
-    waveform_type = [AnalogWaveform]
-
-    for waveform_type, waveform_meta_info in zip(waveform_type, waveform_type_meta_info):
-        waveform = waveform_type()
-        waveform.meta_info = waveform_meta_info()
-        format_name = f"golden_{waveform}.wfm"
-        waveform_dir = f"{Path(__file__).parent}/waveforms"
-        format_path = f"{waveform_dir}/{format_name}"
-
-        waveform_name = "test_format.wfm"
-        waveform_path = f"{waveform_dir}/{waveform_name}"
+    waveform = waveform_type()
+    waveform.meta_info = waveform_meta_info()
+    format_name = f"golden_{waveform}.wfm"
+    waveform_dir = f"{Path(__file__).parent}/waveforms"
+    golden_path = f"{waveform_dir}/{format_name}"
 
-        if os.path.isfile(os.path.join(os.getcwd(), waveform_path)):
-            os.remove(waveform_path)
+    waveform_path = tmp_path / "test_format.wfm"
 
-        values = np.array([10, 11, 12, 32222, 32223, 32224], dtype=np.int16)
+    values = np.array([10, 11, 12, 32222, 32223, 32224], dtype=np.int16)
 
-        waveform.y_axis_values = values
-        waveform.y_axis_spacing = 1 / type_max(np.dtype(np.int16))
-        write_file(waveform_path, waveform)
+    waveform.y_axis_values = values
+    waveform.y_axis_spacing = 1 / type_max(np.dtype(np.int16))
+    waveform.trigger_index = values.size / 2
```

Comment:
@tekperson, does this seem reasonable to do? To set the trigger point at the halfway mark?

Reply:
He mentioned to me that it was what is expected